### DISCUSSION TITLE
Reformat Code

### DISCUSSION THREAD

**Discussion by fedaalgo** [posit-dev/positron#4171] [Feedback]
hi there folks, i loved using Positron right away! however, i've been struggling to find something for reformatting code like the one from the Code Tools on RStudio; is there a way for doing this? maybe with a particular extension?

![Screenshot from 2024-07-29 10-19-24](https://github.com/user-attachments/assets/5a60483a-5b61-40f8-8117-da0d861c8881)

thanks a lot for your time

**Comment by juliasilge** [comment]
Right now, we have the beginnings of a formatter for R, using [styler](https://styler.r-lib.org/). You can access it from the command palette, and you can either "Format Document" or "Format Selection" (there are keyboard shortcuts for both):

![Screenshot 2024-07-29 at 11 48 08 AM](https://github.com/user-attachments/assets/4e6f8e14-6dfc-444b-a083-24ba3f72c192)

Be aware that, like I said, it uses styler and we have plans for a more powerful, not-from-R based formatter over the coming months.

**Reply by fedaalgo** [reply]
that's awesome, thanks a lot!
